docs/topics/development/building_and_running_services.md

```diff
@@ -0,0 +1,98 @@
+# Building and Running Services
+
+## Dockerfile Details
+
+The Dockerfile for the **addons-server** project uses a multi-stage build to optimize the image creation process. Here’s an overview of the key concepts and design decisions behind it:
+
+1. **Multi-Stage Build**:
+   - **Intent**: Multi-stage builds allow Docker to parallelize steps that don't depend on each other and to better cache between layers. This results in more efficient builds by reducing the size of the final image and reusing intermediate layers.
+   - **Layer Caching**: The use of `--mount=type=cache` arguments helps cache directories across builds, particularly useful for node and pip dependencies, dramatically speeding up future builds.
+
+2. **OLYMPIA_USER**:
+   - **Creating a Non-Root User**: The Dockerfile creates an `olympia` user to run the application. This allows the container to run processes as a non-root user, enhancing security by preventing privilege escalation.
+   - **Why Non-Root?**: Running containers as root is considered an antipattern for Python projects due to security vulnerabilities. Using a non-root user like `olympia` ensures that even if an attacker accesses the container, they cannot escalate privileges to the host.
+
+3. **Mounts in Docker Compose**:
+   - **Mounting Local Repository**: The volume `.:/data/olympia` mounts the local Git repository into the container, allowing real-time changes to files within the container.
+   - **Mounting Dependencies**: The volume `./deps:/deps` mounts the dependencies directory, enabling better caching across builds and providing visibility for debugging directly on the host.
```

Comment:
This is no longer done by default though (https://github.com/mozilla/addons-server/pull/22332), so maybe make that clear and explain how to achieve that ?

Reply:
Done